app/controllers/devise/passwords_controller.rb

```diff
@@ -77,4 +82,27 @@ def unlockable?(resource)
     def translation_scope
       'devise.passwords'
     end
+
+    # Helper for use after calling send_*_instructions methods on a resource.
+    # If we are in paranoid mode, we always act as
+    # if the resource was valid and active for authentication
+    # and instructions were sent.
+    def successfully_sent?(resource)
```

Comment:
@bogdanvlviv I'm sorry. I've looked the `DeviseController` and the `successfully_sent?` have this same behavior of check and set flash messages. I think we can keep this way and do this refactor in another moment. :)